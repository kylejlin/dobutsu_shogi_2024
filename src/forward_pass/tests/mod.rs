use super::*;

mod helpers;

use helpers::IntoPretty;

#[test]
fn initial_search_node_is_correct() {
    insta::assert_snapshot!(SearchNode::initial().pretty());
}

#[test]
fn initial_search_node_allegiance_inversion_is_correct() {
    insta::assert_snapshot!(SearchNode::initial()
        .into_builder()
        .invert_active_player()
        .pretty());
}

#[test]
fn initial_search_node_with_initialized_next_action() {
    insta::assert_snapshot!(SearchNode::initial()
        .into_builder()
        .init_next_action()
        .pretty());
}

#[test]
fn initial_search_node_horizontally_flipped_is_correct() {
    insta::assert_snapshot!(SearchNode::initial()
        .into_builder()
        .horizontally_flip()
        .pretty());
}

#[test]
fn initial_search_node_partially_built_is_correct() {
    insta::assert_snapshot!(SearchNode::initial()
        .into_builder()
        .build_without_horizontal_normalization()
        .pretty());
}

#[test]
fn initial_search_node_allegiance_inverted_partially_built_is_correct() {
    insta::assert_snapshot!(SearchNode::initial()
        .into_builder()
        .invert_active_player()
        .build_without_horizontal_normalization()
        .pretty());
}

#[test]
fn initial_search_node_children_are_correct() {
    insta::assert_snapshot!(SearchNode::initial().children().pretty());
}

#[test]
fn initial_search_node_child0_children_are_correct() {
    let child0 = SearchNode::initial().children().0[0].pretty();
    let children = child0.0.children().pretty();
    insta::assert_snapshot!(format!("parent:\n{child0}\n\nchildren:\n{children}"));
}
