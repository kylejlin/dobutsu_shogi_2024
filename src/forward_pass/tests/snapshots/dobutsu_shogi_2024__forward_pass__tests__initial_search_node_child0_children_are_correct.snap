---
source: src/forward_pass/tests/mod.rs
expression: "format!(\"parent:\\n{child0}\\n\\nchildren:\\n{children}\")"
---
parent:
|---|                |---|
|G*E|                |GLE|
|*CL|                |*C*|
|*c*|                |lc*|
|elg|                |e*g|
|---|                |---|
next_action: active_lion to r0c0

children:
SearchNodeSet(len = 4) [
    0:
    |---|                |---|
    |G*E|                |G*E|
    |*CL|                |*CL|
    |lc*|                |lc*|
    |e*g|                |e*g|
    |---|                |---|
    next_action: active_lion to r0c0
    ==========================
    1:
    |---|                |---|
    |G*E|                |G*E|
    |LC*|                |*CL|
    |lc*|                |*cl|
    |e*g|                |e*g|
    |---|                |---|
    next_action: active_lion to r0c0
    ==========================
    2:
    |---|                |---|
    |GLE|                |G*E|
    |***|                |*cL|
    |lC*|                |***|
    |e*g|                |elg|
    |---|                |---|
    next_action: active_lion to r0c0
    ==========================
    3:
    |---|                |---|
    |*LE|                |G*E|
    |GC*|                |*CL|
    |lc*|                |*cg|
    |e*g|                |el*|
    |---|                |---|
    next_action: active_lion to r0c0
    ==========================
]
